In [1]:
import numpy as np
import pandas as pd

### __Question 1__

A factory claims avg. wieight of cereal box is 500 grams. A sample of 64 boxes is taken at random with a mean of 490 grams and population standard deviation is 15 grams (sigma) with significance level at 5%. Determine the validity of this claim using hypothesis testing.

#### __Solution:__

- It is a two-tailed test as it is not mentioned if it is a one-tailed test.
- It is a one-sample test.
- It qualifies for Z-test as sample size > 30 and population standard deviation is known.
- Since, it is a two-tailed test, Z-score will be (α / 2) = (0.05 / 2) = 0.025.
- So, Z value will be from __-1.96__ to __1.96__. ($Z_{critical}$)

In [2]:
x_bar = 490
mu = 500
sigma = 15
n = 64

Z = (x_bar - mu) / (sigma/np.sqrt(n))
Z

np.float64(-5.333333333333333)

##### __Observation:__

- Here, __Z statistic__ is coming as -5.33 which lies outside the acceptance region (i.e less than) between `-2.576` to `2.576`.
- Hence, we can reject the null hypothesis.
- So, the claim made by the company stating that the average weight of the box being 500 grams is incorrect.

### __Question 2__

A company claims the average battery life of their batteries is 300  hours. A sample of 100 batteries is taken at random and they show average of 295 hours and population standard deviation of 10 hours at significance level of 1%. Determine the validity of this claim using hypothesis testing.

#### __Solution:__

- It is a two-tailed test as it is not mentioned if it is a one-tailed test.
- It is a one-sample test.
- It qualifies for Z-test as sample size > 30 and population standard deviation is known.
- Since, it is a two-tailed test, Z-score will be (α / 2) = (0.01 / 2) = 0.005.
- So, Z value will be __-2.57__ to __2.57__. ($Z_{critical}$)

* $H_0$ = 300
* $H_A$ $\neq$ 300

In [3]:
x_bar = 295
mu = 300
sigma = 10
n = 100

Z = (x_bar - mu) / (sigma/np.sqrt(n))
Z

np.float64(-5.0)

##### __Observation:__

- Here, __Z statistic__ is coming as -5 which lies outside the acceptance region (i.e. less than) between `-2.57` to `2.57`.
- Hence, we can reject the null hypothesis.
- So, the claim made by the company stating that the average battery life of their batteries being 300 hours is incorrect.

### __Question 3__

In a company, there is a Group A which has 40 employees and average productivity of 120 units and standard deviation of 15. Group B has 50 employees with average productivity of 125 units and standard deviation of 20. Is there a significant difference between their productivities at a significance level of 5%.

#### __Solution:__

- It is a two-tailed test as it is not mentioned if it is a one-tailed test.
- It is a two-sample test.
- It qualifies for Z-test as sample size > 30 and population standard deviation is known.
- Since, it is a two-tailed test, Z-score will be (α / 2) = (0.05 / 2) = 0.025.
- So, Z value will be __-1.96__ to __1.96__. ($Z_{critical}$)

##### __Observation:__

- Here, __Z statistic__ is coming as -1.36 which lies inside the acceptance region or confidence interval between `-1.96` to `1.96`.
- Hence, we fail to reject the null hypothesis.
- So, there is no significant difference between the two groups' productivity, i.e. they are working at the same productivity level.

### __Question 4__

Medicine A reduces b.p  by 10 mm/Hg for a sample size of 60 and standabrd deviation at 2. Medicine B reduces b.p  by 12 mm/Hg for a sample size of 70 and standabrd deviation at 2.5. Significant level is 5%.